cheatsheets_draft/Secrets_Management_CheatSheet.md

```diff
@@ -401,33 +401,107 @@ EaaS is a model in which users subscribe to a cloud-based encryption service wit
+- Consider enabling secrets detection at the developer level to avoid checking secrets into code before commit/PR either in the IDE, as part of test-driven development, or via pre-commit hook.
+- Make secrets detection part of the threat model. Consider secrets as part of the attack surface during threat modeling exercises.
+- Evaluate detection utilities and related signatures often to ensure they meet expectations.
+- Consider having more than one detection utility and correlating/de-duping results to identify potential areas of detection weakness.
+- Also consider exploring a balance between entropy and ease of detection. Secrets with consistent formats are easier to detect with lower false-positive rates, but you also don't want to miss a human-created password simply because it doesn't match your detection rules.
+
+
+### 8.2 Types of secrets to be detected
+
+Many types of secrets exists and signatures for each should be considered to ensure accurate detection for all. Among the more common types are:
+
+- High availability secrets (Tokens that are difficult to rotate)
+- Application configuration files
+- Connection strings
+- API keys
+- Credentials
+- Passwords
+- 2FA keys
+- Private keys (like ssh keys)
+- Session tokens
+- Platform specific secret types (e.g.; Amazon Web Services, Google Cloud)
+
+For more fun learning about secrets and practice rooting them out check out the [Wrong Secrets](https://owasp.org/www-project-wrongsecrets/) project.
+
+### 8.3 Detection lifecycle
+
+Secrets are like any other authorization token. They should:
+
+- Exist only for as long as necessary (rotate often)
+- Have a method for automatic rotation
+- Only be visible to those who need them (least privilege)
+- Be revokable (including the logging of attempt to use a revoked secret)
+- Never be logged (must implement either an encryption or masking approach in place to avoid logging plaintext secrets)
+
+Detections should exist during each of these stages of the secret lifecycle to ensure secrets were not introduced but missed in a prior step.
+
+### 8.4 Documentation for how to detect secrets
+
+Ensure documentation is created and updated regularly to inform the developer community on procedures and systems available at your company and what types of secrets management is expected, how to test for secrets, and what to do in event of secrets detected.
+
+Documentation should:
+
+- Exist and be updated often, especially in response to an incident
+- Include the following information:
+    - Who has access to the secret
+    - How it gets rotated
+    - Any upstream or downstream dependencies that could potentially be broken during secret rotation
+    - Who is the point of contact during an incident
+    - Security impact of exposure
+
+- Identify when secrets may be handled differently depending on the threat risk, data classification, etc.
+
+## 9 Incident Response
```

Comment:
Could you edit the index up top to match this title?

Reply:
Will do in final pr